infrastructure-cardiac.md

```diff
@@ -0,0 +1,63 @@
+# Cardiac Electrophysiology Web Lab infrastructure
+
+This document describes the infrastructure that is specific to the Cardiac Electrophysiology Web Lab.
+The generic Web Lab infrastructure is defined [here](./infrastructure.md).
+
+## Models
+
+Models are implemented in CellML.
+
+In the future, they will be converted to [SymPy](https://www.sympy.org/en/docs.html) expressions using [cellmlmanip](https://github.com/ModellingWebLab/cellmlmanip) (with units handled in [pint](https://pint.readthedocs.io)).
+The resulting 
+
+## Protocols
+
+Protocols are written in functional curation syntax (see below).
+
+## Functional curation
+
+Web Lab experiments run `Functional curation` with a model and protocol.
+
+### FC-Runner
+
+The Celery task to run experiments this is stored in the [fc-runner](https://github.com/ModellingWebLab/fc-runner/tree/master/fcws) repository.
+The task checks input from the web interface, and fires off the `fc` experiment.
+When completed, it packs up the results and sends them back to the web interface.
+
+**Should we rename it to `weblab-fc-runner` ??**
+
+### FC
+
+The actual experiment running code is stored in the [weblab-fc](https://github.com/ModellingWebLab/weblab-fc) repository.
+
+This repo contains the Python `fc` module, which is at the heart of the cardiac EP Web Lab.
+It:
+
+- Parses CellML models.
+  - Currently this is handled by PyCml, but we are converting to [cellmlmanip](https://github.com/ModellingWebLab/cellmlmanip).
+- Parses FC protocols (using fc's `CompactSyntaxParser` module)
+- Checks model and protocol compatibility
+- Manipulates the model, e.g. adding/removing variables or stimulus protocols
+  - Again should transition from PyCml to cellmlmanip
+- Generates model-specific Python code (extending the `fc` `AbstractModel` class) that can run simulations.
+  - Currently this is handled by PyCml, but we are converting to [weblab-cg](https://github.com/ModellingWebLab/weblab-cg).
+  - Simulations are run in CVODE, using Cython to interface with it.
+- Executes the protocol
+
+
+**Concepts that need explaining:**
+
+- Expressions/statements/values
+- Protocol
+- Environment
+- Ranges (why are these in `fc.simulations` rather than `fc.language` ?)
+
+## Web Lab without FC?
+
+At the start of the project, we discussed allowing arbitrary Python scripts, running in some kind of sandbox.
```

Comment:
This was not so much 'without FC' but extending FC to allow Python for post-processing etc not just the domain-specific language.